### PR TITLE
implement vmode storing and loading

### PR DESCRIPTION
**Pull request by apatlpo**
@NoeLahaye , please let me know what you think about this.
This will be directly relevant to your work (as well as mine of course)

### CODE REVIEW

**Comment by apatlpo** [comment]
The notebook `vmodes.ipynb` contains an example, see cells 7 and 13

**Comment by NoeLahaye** [comment]
Why using two separate files ? Because they involve different datasets?

File: crocosi/vmodes.py [hunk 1]
```diff
@@ -435,6 +439,87 @@ def _w2rho(self, data, zf=None, zc=None, align=True):
...
+    
+    Parameters
+    ----------
+    file_path: str
+        Path to vmode datafile (zarr archive)
+    xgrid: xgcm.Grid object
+        Required for grid manipulations
+    """
+    ds = xr.open_zarr(file_path)
+    vm = Vmodes(xgrid,
```
[the remark this answers is not in the document]
Comment (apatlpo) [reply crocosi/vmodes.py:494]: I don't think that they would be computed if `persist=False` but will check

[the remark this answers is not in the document]
Comment (NoeLahaye) [reply crocosi/vmodes.py:494]: OK, but then I don't see were the modes are actually copied (from the zarr dataset) into the vmodes object?

File: crocosi/vmodes.py [hunk 2]
```diff
@@ -435,6 +439,87 @@ def _w2rho(self, data, zf=None, zc=None, align=True):
...
+        if projections:
+            _file = file_path.strip('.zarr')+'_projections.zarr'
+            ds = _move_singletons_as_attrs(projections)
+            ds.to_zarr(_file, **kwargs)
+            print('Store projections in {}'.format(_file))
+    
+    def _wrap_in_dataset(self):
+        """ wrap critical data in a xarray dataset
+        """
+        # we won't store xgrid as the intend is to store this in a zarr archive
```
[the remark this answers is not in the document]
Comment (apatlpo) [reply crocosi/vmodes.py:474]: hum ... I'm not sure I want to / can go down that road quite now.
Let's leave like that for now.
I'll add your suggestion as a comment in the code to remember we may be able to do that.

File: crocosi/vmodes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (apatlpo) [reply crocosi/vmodes.py:494]: ok I think I checked this.

**Comment by apatlpo** [comment]
Good to merge?

File: crocosi/vmodes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (NoeLahaye) [reply crocosi/vmodes.py:492]: I still don't see whether the modes stored in zarr are copied into (or assigned to) `vm.ds` ?

[the remark this answers is not in the document]
Comment (apatlpo) [reply crocosi/vmodes.py:498]: sorry I didn't see you earlier comment. Off course you're right !
Hopefully da7ce22d323dddcc5d583b497e7b9a79078d564f fixes this.

**Comment by NoeLahaye** [comment]
Great, thanks, good to merge.

**Comment by apatlpo** [comment]
ok, thanks.

Sorry, I just saw you' re comments about using just one file.
It's possible but would require more book keeping.
Let's see first how it works out with two files.
Also remember we could use `store/load_diagnostic` from `postp`objects for projections.

merging

**Comment by apatlpo** [comment]
let's work on using this now !!!

( ... and discovering bugs :) )

**Pull request merged**